src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -406,10 +406,11 @@ public async Task InvokeAsync(ScriptInvocationContext invocationContext)
             // This could throw if no initialized workers are found. Shut down instance and retry.
             IEnumerable<IRpcWorkerChannel> workerChannels = await GetInitializedWorkerChannelsAsync(invocationContext.FunctionMetadata.Language ?? _workerRuntime);
             var rpcWorkerChannel = _functionDispatcherLoadBalancer.GetLanguageWorkerChannel(workerChannels);
-            if (rpcWorkerChannel.FunctionInputBuffers.TryGetValue(invocationContext.FunctionMetadata.GetFunctionId(), out BufferBlock<ScriptInvocationContext> bufferBlock))
+            string functionId = invocationContext.FunctionMetadata.GetFunctionId();
```

Comment:
Can you update line 203 to 
```csharp
&& webhostLanguageWorkerChannel.Any(a => a.Value.Task.Status == TaskStatus.RanToCompletion))
```

Replacing Where + Any to just Any.

Reply:
Also 547 to 
```csharp
if (!(await GetInitializedWorkerChannelsAsync()).Any())
```

Reply:
Made those changes in addition to a couple of other improvements I was tracking.